src/devices/bus/a2bus/suprterminal.cpp

```diff
@@ -173,14 +164,36 @@ void a2bus_suprterminal_device::write_c0nx(u8 offset, u8 data)
 		case 6:
 			m_bCharBank1 ^= 1;
 			break;
+	}
+}
 
-		case 8:
-			m_crtc->address_w(data);
-			break;
+u8 a2bus_suprterminal_device::read_c0nx(u8 offset)
+{
+	if (offset < 8)
+	{
+		access_c0nx(offset);
+	}
+	else if (offset == 9)
+	{
+		return m_crtc->register_r();
+	}
```

Comment:
You need to check `machine().side_effects_diabled()` before doing things with side effects in response to a read.  If you don’t, then having a debugger memory window open will wreak havoc with the system.

Reply:
It looks like this check is already done in `apple2_state::c080_r`.

Reply:
Having the check there is a problem in and of itself, as it means you can’t see what would be read from the C0nx range when debugging without forcing a read with side effects.

The slot cards really should all be audited for doing the checks themselves so the check can be removed from the host system and the debugger can become more useful.  Adding more cards without checks means more work later.

Reply:
Okay, makes sense. Updated.